### ISSUE TITLE
mapPolygon() not plotting coastline object correctly

### ISSUE THREAD

**Issue by clayton33** [dankelley/oce#1181]
The attached code shows an example of using `plot()` and plotting a coastline object using `polygon()` (left panel) which plots the polygons properly. The middle panel uses `mapPlot()` and `mapPolygon()`, where the polygons are not plotted correctly. The right panel uses `mapPlot()` and a few lines of code from `mapPolygon()` which plots the polygon's correctly. 

The original file used is a `.shp` file, which is read using `read.coastline.shapefile()`. The example coastline object below, `cl` , reflects the structure of the file when read. There appears to be something funny going on with `mapPolygon()`?

```
library(oce)
library(ocedata)
data(coastlineWorldFine)

cl <- as.coastline(longitude = c(NA, -45.819, -45.881, - 45.842, -45.842, -45.780, NA, 
                                 -46.100, -46.302, -46.302, -46.100, NA, -46.620, -46.653, -45.212,
                                 -45.149, -45.455, -46.142),
                   latitude = c(NA, 47.048, 47.048, 47.196, 47.359, 47.358, NA, 45.819, 45.819, 
                                   45.996, 45.996, NA, 48.314, 48.353, 49.002, 48.941, 48.827, 48.480))

latlim <- c(46,49)
lonlim <- c(-47,-44)

par(mfrow=c(1,3))
plot(coastlineWorldFine, clatitude= 47.5, clongitude= -45.5, span=700, mar=c(2,2,1,1))
polygon(cl[['longitude']], cl[['latitude']], col='grey')

mapPlot(coastlineWorldFine, latitudelim = latlim, longitudelim=lonlim, 
               proj="+proj=merc", col='lightgray')
mapPolygon(cl[['longitude']], cl[['latitude']])

mapPlot(coastlineWorldFine, latitudelim = latlim, longitudelim=lonlim, 
               proj="+proj=merc", col='lightgray')
xy <- lonlat2map(cl[['longitude']], cl[['latitude']])
polygon(xy$x, xy$y, col='grey')
```

**Comment by dankelley** [comment]
Thanks. I made a test suite from your excellent test code (https://github.com/dankelley/oce-issues/tree/master/11xx/1181) and it produces as below, clearly demonstrating an error.
![1181a](https://cloud.githubusercontent.com/assets/99469/22292560/47f5bac8-e2e2-11e6-9b68-abbcae14f1dc.png)

**Comment by dankelley** [comment]
By commenting out some code, I see that the problem is in the use of `badFillFix1()` and/or `badFillFix2()`, called in lines 2415 and 2416 of `R/map.R'.  These are tricky functions designed to help with the ugly-horizontal-line problem.  I'm a bit scared to just drop them completely, although I am not entirely sure we even need them anymore since `coastlineCut()` seems to work reasonably well nowadays. Everything to do with u-h-l is tricky, and these functions are no exception. It may take a bit of study and testing to get a solution. I'll look into this sometime later today, likely this evening.

**Comment by clayton33** [comment]
Thanks, I had a feeling the badFillFix functions had to do with the UHL problem.

**Comment by dankelley** [comment]
This is possibly fixed in `develop` commit cf09d12ac5513407ae0960ec9cc2429fccfc7bcd. Can @clayton33 check the results and take a quick look at the code to see what I did ... and then close the issue if it seems OK?  (If you click on the link for the commit, it gets you to a diff on the code, so you can see what I did.)
